### ISSUE TITLE
Shipping and Payment pages

### ISSUE THREAD

**Issue by DivaVocals** [DivaVocals/zen_COWOA#16]
So I think it's time that this module let's go of the old Zen Cart/osCommerce thinking that shipping and payment needs to be on two separate pages.. Thinking that a GOOD long term direction for COWOA is to combine the shipping and payment steps on one page.

AND/OR 

add some express checkout functionality to COWOA

Perhaps it's time that this module (and even Zen Cart in general) needs to support the notion of "expedited" checkout. What I mean behind expedited checkout is similar to how PayPal Express works if you turn on ALL of it's features.. For example:
If there is only ONE shipping option or ONE payment option (**excluding** PayPal and credit card payments) available for this order, then proceed directly to the order confirmation page. -- In the past the prevailing train of though was that bypassing the shipping and/or payment pages under ANY circumstances was BAD because what if the shopper wanted to change the shipping and/or billing address?? Except you are provided the SAME method to change either from the checkout confirmation page.
With regards to shipping, express checkout functionality would need requirements as follows:
If there is only one shipper, but more than one class of shipping is available for this shipper, DO NOT bypass the shipping page. 
If there is only one shipper, and only ONE class of shipping is active, (ie: USPS with ONLY Priority Mail active), then bypass the shipping page.

With regards to payments, express checkout functionality would be pretty limited, and not sure if this would even be beneficial to any but a very SMALL group of shopowners. However if active need requirements as follows:
For shops that do not have PayPal or credit card or other third party payment processing active, customer should be given the option to choose express checkout at the beginning of the checkout process (which is the same place they must choose PayPal Express checkout if they wish). 

Not sure there's any REAL benefit to bypassing the payment page except in some very narrow circumstances. For one thing it leaves out the process of entering coupons and GVs.

**Comment by davesaddle** [comment]
Agree...This is how I always did mine..  basically, using cowoa or standard checkout would take you to user form, shopping cart on the top, billing info with a button for shipping address on the left.  if the shipping address button was hit, it would expand to display the shipping form.  if not pushed, billing address was auto coped to shipping fields when processed in the header page.  confidence boxes on the right.

next page... again showed cart on top, shipping options on the left, payment options on the right, comment and redeemed boxes on the bottom.  picking a shipment or payment with a fee would auto update cart totals while user was still on the page.

next page, confirmation, cart so on...  making a three page checkout... found that this config left it responsive and usable on phone screens.

one page would be to much scrolling for phones, but may work by making the non-default options expandable like the shipping address.  Also the re-configuring forms to ninja code and using front awesome makes the forms supper compact.

I was about to start coding this up on my site design, which has the responsive blue back-end.... 
This adds more observers and some pages for the cart... should't add anything to the sql or tables.  I was starting to strip out the pages/code from my old site the other day to start the process.

The old FEC was the idea, but not all the code...

**Comment by DivaVocals** [comment]
"one page would be to much scrolling for phones, but may work by making the non-default options expandable like the shipping address."

"one page checkout won’t necessarily perform better than a comparatively optimized multi-step checkout"... 

"One page check out" is smoke and mirrors in my opinion.. It is what people strive for when they are too lazy to look at the whole process and improve it. It also doesn't "save steps" or even a significant amount of time.. One page checkout is also overwhelming on a mobile device.. Probably why you don't see big box stores using them in their mobile versions.. 

Putting the shipping and billing on ONE page makes LOGICAL sense.. I have tried to do this before, but because development is not a strong skillset of mine, I've not been successful yet.. So it would be MAD cool if you could help in this regard..

and I agree..

**Comment by jeking928** [comment]
Agree completely >>> "One page check out" is smoke and mirrors in my opinion

Combining shipping and billing does make sense as does skipping pages when only one option is active, for whatever reason.

I can't do much to help with the coding, but can certainly test.
